numpyro/distributions/copula.py

```diff
@@ -0,0 +1,164 @@
+        cdf = Normal().cdf(normal_samples)
+        return self.marginal_dist.icdf(cdf)
+
+    @validate_sample
+    def log_prob(self, value):
+        # Ref: https://en.wikipedia.org/wiki/Copula_(probability_theory)#Gaussian_copula
+        # see also https://github.com/pyro-ppl/numpyro/pull/1506#discussion_r1037525015
+        marginal_lps = self.marginal_dist.log_prob(value)
+        probs = self.marginal_dist.cdf(value)
+        quantiles = Normal().icdf(clamp_probs(probs))
+
+        copula_lp = (
+            self.base_dist.log_prob(quantiles)
+            + 0.5 * (quantiles**2).sum(-1)
+            + 0.5 * jnp.log(2 * jnp.pi) * quantiles.shape[-1]
+        )
+        return copula_lp + marginal_lps.sum(axis=-1)
+
+    @property
+    def mean(self):
+        return jnp.broadcast_to(self.marginal_dist.mean, self.shape())
+
+    @property
+    def variance(self):
+        return jnp.broadcast_to(self.marginal_dist.variance, self.shape())
+
+    @constraints.dependent_property(is_discrete=False, event_dim=1)
+    def support(self):
+        return constraints.independent(self.marginal_dist.support, 1)
+
+    @lazy_property
+    def correlation_matrix(self):
+        return self.base_dist.covariance_matrix
+
+    @lazy_property
+    def correlation_cholesky(self):
+        return self.base_dist.scale_tril
+
+    def tree_flatten(self):
+        marginal_flatten, marginal_aux = self.marginal_dist.tree_flatten()
+        return (marginal_flatten, self.base_dist.scale_tril), (
+            type(self.marginal_dist),
+            marginal_aux,
+        )
+
+    @classmethod
+    def tree_unflatten(cls, aux_data, params):
+        marginal_flatten, correlation_cholesky = params
+        marginal_cls, marginal_aux = aux_data
+        marginal_dist = marginal_cls.tree_unflatten(marginal_aux, marginal_flatten)
+        return cls(marginal_dist, correlation_cholesky=correlation_cholesky)
+
+
+class GaussianCopulaBeta(GaussianCopula):
+    arg_constraints = {
+        "concentration1": constraints.positive,
+        "concentration0": constraints.positive,
+        "correlation_matrix": constraints.corr_matrix,
+        "correlation_cholesky": constraints.corr_cholesky,
+    }
+    support = constraints.independent(constraints.unit_interval, 1)
+
+    def __init__(
+        self,
+        concentration1,
+        concentration0,
+        correlation_matrix=None,
+        correlation_cholesky=None,
+        *,
+        validate_args=False,
+    ):
+        # set initially to allow argument validation
+        self.concentration1, self.concentration0 = concentration1, concentration0
+
+        super().__init__(
+            Beta(concentration1, concentration0),
+            correlation_matrix,
+            correlation_cholesky,
+            validate_args=validate_args,
+        )
+        self.concentration1, self.concentration0 = promote_shapes(
+            concentration1, concentration0, shape=self.batch_shape + self.event_shape
+        )
```

Comment:
@fehiepsi I cannot say with certainty whether this shape promotion is necessary and it seems strange that we now have two sets of concentration factors (one inside the Beta marginal and one stored in the class itself).

Reply:
You can remove them. They're not used to compute anything here

Reply:
Okay, thanks!